# Estimando $\beta$ de los activos

<img style="float: right; margin: 0px 0px 15px 15px;" src="https://upload.wikimedia.org/wikipedia/commons/4/48/Capital_market_line_of_CAPM.svg" width="400px" height="400px" />

En la clase anterior aprendimos 
- ¿qué es el CAPM?; 
- ¿cuáles son los supuestos sobre los que se funda el CAPM?;
- derivamos la fórmula del CAPM; y
- aprendimos como obtener la $\beta$ de un portafolio a partir de la $\beta$ de activos individuales.

En la clase de hoy estudiaremos una forma de estimar los $\beta$ de activos individuales.

**Objetivos:**
- Revisitar riesgo sistemático y no sistemático.
- Estudiar un método para estimar las $\beta$ de los activos.

*Referencia:*
- Notas del curso "Portfolio Selection and Risk Management", Rice University, disponible en Coursera.
- [Notas del curso "Financial Engineering", Columbia University](http://www.columbia.edu/~ks20/FE-Notes/FE-Notes-Sigman.html)
___

## <font color=green> Conclusiones de la [lectura](https://web.stanford.edu/~wfsharpe/art/parable/parable.htm). </font>

## 1. Riesgo sistemático y no sistemático.

Recordamos la fórmula de CAPM:

$$E[r_i]-r_f=\beta_i(E[r_M]-r_f),$$

donde $\beta_i=\frac{\sigma_{M,i}}{\sigma_M^2}$ y $\sigma_{M,i}$ es la covarianza del portafolio de mercado con el activo individual $i$.

Todas las anteriores son variables determinísiticas.

- ¿Qué pasa si usamos el CAPM como un modelo de rendimientos? Es decir,

$$r_i=r_f+\beta_i(r_M-r_f)+\epsilon_i,$$

donde $\epsilon_i$ es un término de error.

<font color=green> Despejando $\epsilon_i$, tenemos que: </font>

- $E[\epsilon_i]=0$, y
- $cov(\epsilon_i,r_M)=0$.

<font color=green> Ver en el tablero. </font>

Entonces, la varianza del activo $i$ es:

$$\sigma_i^2=\beta_i^2\sigma_M^2+var(\epsilon_i),$$

donde el primer término corresponde al riesgo sistemático (de mercado) y el segundo al riesgo idiosincrático.
___

## 2. Estimando $\beta$ para un activo.

- En el mercado real, la cantidad de activos es ENORME, y tratar de construir el portafolio de mercado sería una tarea grandiosa, pero poco realista para un analista financiero. 
- Por lo tanto, los índices de mercado han sido creados para intentar aproximar el portafolio de mercado.
- Dicho índice es un portafolio más pequeño que el de mercado, construido por los que se consideran los activos más dominantes, y que capturan la esencia de el portafolio de mercado.

- El índice de mercado más conocido es el Standard & Poor’s 500-stock index (S&P), compuesto de 500 activos.
- Un $\beta$ para un activo dado, se puede estimar utilizando el S&P en reemplazo de M, y utilizando datos históricos para ambos rendimientos (el del activo y el del S&P500).
- Por ejemplo, considere un activo $i$ para el cual queremos estimar su $\beta_i$.
- Este estimado se construye usando medias, varianzas y covarianzas muestrales como sigue:
 - Escogemos $N$ rendimientos históricos, tales como los reportados mensualmente hace tres años.
 - Para $k=1,2,\dots,N$, $r_{ik}$ y $r_{S\&Pk}$ denotan el $k-$ésimo valor muestral de los rendimietos.

Entonces

$$\hat{E[r_i]}=\frac{1}{N}\sum_{k=1}^{N}r_{ik}, \text{ y}$$

$$\hat{E[r_{S\&P}]}=\frac{1}{N}\sum_{k=1}^{N}r_{S\&Pk}.$$

Además, la varianza $\sigma_{S\&P}^2$ se estima como

$$\hat{\sigma_{S\&P}^2}=\frac{1}{N-1}\sum_{k=1}^{N}(r_{S\&Pk}-\hat{E[r_{S\&P}]})^2,$$

y la covarianza $\sigma_{S\&P,i}$

$$\hat{\sigma_{S\&P,i}}=\frac{1}{N-1}\sum_{k=1}^{N}(r_{S\&Pk}-\hat{E[r_{S\&P}]})(r_{ik}-\hat{E[r_i]})$$

Finalmente, 

$$\hat{\beta_i}=\frac{\hat{\sigma}_{S\&P,i}}{\hat{\sigma}_{S\&P}^2}.$$

### Ejemplo...

Entrar a yahoo finance, a la información de MSFT, TSLA y ^GSPC.

In [1]:
# Importar paquetes
import pandas_datareader.data as web
import yfinance as yf
import pandas as pd
import numpy as np
%matplotlib inline

yf.pdr_override()

In [5]:
# Función para descargar precios de cierre ajustados:
def get_adj_closes(tickers,
                   start_date=None,
                   end_date=None,
                   interval="1d"
                  ):
    # Fecha inicio por defecto (start_date='2010-01-01') y fecha fin por defecto (end_date=today)
    # Descargamos DataFrame con todos los datos
    closes = web.get_data_yahoo(
        tickers=tickers,
        start=start_date,
        end=end_date,
        interval=interval
    )
    # Solo necesitamos los precios ajustados en el cierre
    closes = closes['Adj Close']
    # Se ordenan los índices de manera ascendente
    closes.sort_index(inplace=True)
    return closes

In [6]:
# Importar datos de MSFT, TSLA, y ^GSPC
tickers = ["MSFT", "TSLA", "^GSPC"]
start = "2018-04-01"

In [10]:
closes = get_adj_closes(tickers, start_date=start, interval="1mo")

[*********************100%***********************]  3 of 3 completed


In [11]:
closes

MSFT        TSLA        ^GSPC
Date                                                          
2018-04-01 00:00:00-04:00   88.261810   19.593332  2648.050049
2018-05-01 00:00:00-04:00   93.282707   18.982000  2705.270020
2018-06-01 00:00:00-04:00   93.468987   22.863333  2718.370117
2018-07-01 00:00:00-04:00  100.549583   19.875999  2816.290039
2018-08-01 00:00:00-04:00  106.473709   20.110666  2901.520020
...                               ...         ...          ...
2022-12-01 00:00:00-05:00  239.220825  123.180000  3839.500000
2023-01-01 00:00:00-05:00  247.190857  173.220001  4076.600098
2023-02-01 00:00:00-05:00  248.796829  205.710007  3970.149902
2023-03-01 00:00:00-05:00  288.299988  207.460007  4109.310059
2023-04-01 00:00:00-04:00  283.489990  180.539993  4091.949951

[61 rows x 3 columns]

In [12]:
# Obtener rendimientos mensuales
ret = closes.pct_change().dropna()

In [13]:
ret

MSFT      TSLA     ^GSPC
Date                                                   
2018-05-01 00:00:00-04:00  0.056886 -0.031201  0.021608
2018-06-01 00:00:00-04:00  0.001997  0.204474  0.004842
2018-07-01 00:00:00-04:00  0.075753 -0.130660  0.036022
2018-08-01 00:00:00-04:00  0.058917  0.011807  0.030263
2018-09-01 00:00:00-04:00  0.022079 -0.122290  0.004294
2018-10-01 00:00:00-04:00 -0.066101  0.274011 -0.069403
2018-11-01 00:00:00-04:00  0.038199  0.039013  0.017859
2018-12-01 00:00:00-05:00 -0.080090 -0.050445 -0.091777
2019-01-01 00:00:00-05:00  0.028158 -0.077464  0.078684
2019-02-01 00:00:00-05:00  0.072776  0.041886  0.029729
2019-03-01 00:00:00-05:00  0.057250 -0.125109  0.017924
2019-04-01 00:00:00-04:00  0.107342 -0.147109  0.039313
2019-05-01 00:00:00-04:00 -0.052986 -0.224266 -0.065778
2019-06-01 00:00:00-04:00  0.087127  0.206848  0.068930
2019-07-01 00:00:00-04:00  0.017244  0.081223  0.013128
2019-08-01 00:00:00-04:00  0.011668 -0.066222 -0.018092
2019-09-01 00:00:00-04:00  0.011845  0.067639  0.017181
2019-10-01 00:00:00-04:00  0.031216  0.307427  0.020432
2019-11-01 00:00:00-04:00  0.055870  0.047695  0.034047
2019-12-01 00:00:00-05:00  0.045294  0.267897  0.028590
2020-01-01 00:00:00-05:00  0.079455  0.555160 -0.001628
2020-02-01 00:00:00-05:00 -0.048288  0.026776 -0.084110
2020-03-01 00:00:00-05:00 -0.023883 -0.215557 -0.125119
2020-04-01 00:00:00-04:00  0.136326  0.492137  0.126844
2020-05-01 00:00:00-04:00  0.022543  0.067939  0.045282
2020-06-01 00:00:00-04:00  0.113652  0.293186  0.018388
2020-07-01 00:00:00-04:00  0.007371  0.325011  0.055101
2020-08-01 00:00:00-04:00  0.100093  0.741452  0.070065
2020-09-01 00:00:00-04:00 -0.065142 -0.139087 -0.039228
2020-10-01 00:00:00-04:00 -0.037370 -0.095499 -0.027666
2020-11-01 00:00:00-04:00  0.057293  0.462736  0.107546
2020-12-01 00:00:00-05:00  0.041726  0.243252  0.037121
2021-01-01 00:00:00-05:00  0.042892  0.124506 -0.011137
2021-02-01 00:00:00-05:00  0.001811 -0.148740  0.026091
2021-03-01 00:00:00-05:00  0.016925 -0.011207  0.042439
2021-04-01 00:00:00-04:00  0.069602  0.062147  0.052425
2021-05-01 00:00:00-04:00 -0.009914 -0.118713  0.005486
2021-06-01 00:00:00-04:00  0.087494  0.087137  0.022214
2021-07-01 00:00:00-04:00  0.051717  0.011034  0.022748
2021-08-01 00:00:00-04:00  0.059563  0.070605  0.028990
2021-09-01 00:00:00-04:00 -0.064331  0.054042 -0.047569
2021-10-01 00:00:00-04:00  0.176291  0.436530  0.069144
2021-11-01 00:00:00-04:00 -0.003106  0.027612 -0.008334
2021-12-01 00:00:00-05:00  0.019194 -0.076855  0.043613
2022-01-01 00:00:00-05:00 -0.075345 -0.113609 -0.052585
2022-02-01 00:00:00-05:00 -0.039199 -0.070768 -0.031360
2022-03-01 00:00:00-05:00  0.033995  0.238009  0.035773
2022-04-01 00:00:00-04:00 -0.099867 -0.191945 -0.087957
2022-05-01 00:00:00-04:00 -0.020359 -0.129197  0.000053
2022-06-01 00:00:00-04:00 -0.053120 -0.111888 -0.083920
2022-07-01 00:00:00-04:00  0.093097  0.323765  0.091116
2022-08-01 00:00:00-04:00 -0.068640 -0.072489 -0.042440
2022-09-01 00:00:00-04:00 -0.107376 -0.037589 -0.093396
2022-10-01 00:00:00-04:00 -0.003306 -0.142168  0.079863
2022-11-01 00:00:00-04:00  0.099126 -0.144326  0.053753
2022-12-01 00:00:00-05:00 -0.057397 -0.367334 -0.058971
2023-01-01 00:00:00-05:00  0.033317  0.406235  0.061753
2023-02-01 00:00:00-05:00  0.006497  0.187565 -0.026112
2023-03-01 00:00:00-05:00  0.158777  0.008507  0.035052
2023-04-01 00:00:00-04:00 -0.016684 -0.129760 -0.004225

In [14]:
# Obtener matriz de covarianza
sigma = ret.cov()
sigma

MSFT      TSLA     ^GSPC
MSFT   0.004067  0.007025  0.002683
TSLA   0.007025  0.047980  0.005857
^GSPC  0.002683  0.005857  0.002893

In [15]:
# Beta de Microsoft
beta_msft = sigma.loc["MSFT", "^GSPC"] / sigma.loc["^GSPC", "^GSPC"]
beta_msft

0.9271466455597361

In [16]:
# Beta de Tesla
beta_tsla = sigma.loc["TSLA", "^GSPC"] / sigma.loc["^GSPC", "^GSPC"]
beta_tsla

2.0243484751074803

## 3. Ejercicios

Su portafolio de inversión consiste de $\$150,000$ invertidos en un solo activo. Suponga que la tasa de interés libre de riesgo es $3\%$. Este activo tiene un rendimiento esperado del $12\%$, y una volatilidad del $40\%$, y el portafolio de mercado tiene un rendimiento esperado del $10\%$ y una volatilidad del $18\%$. Bajo las suposiciones del CAPM, ¿cuál es la volatilidad de un mejor portafolio alternativo que tiene el mismo rendimiento que el activo?

A. $\sigma = 15.52\%$

**B. $\sigma = 23.14\%$**

C. $\sigma = 30.25\%$ 

D. $\sigma = 35.13\%$ 

Con los mismos datos del ejercicio anterior, y suponiendo que usted está conforme con el riesgo del activo, bajo las suposiciones de CAPM, ¿cuál sería un mejor rendimiento esperado que debería ganar?

A. $E[r] = 13.65\%$

B. $E[r] = 15.30\%$

**C. $E[r] = 18.56\%$**

D. $E[r] = 20.38\%$ 

Suponga que usted tiene invertidos $\$30,000$ en los siguientes cuatro activos:

| Activo        | Cantidad invertida | $\beta$ |
| ------------- | ------------------ | ------- |
| Activo A      | $\$5,000$          | 0.75    |
| Activo B      | $\$10,000$         | 1.10    |
| Activo C      | $\$8,000$          | 1.36    |
| Activo D      | $\$7,000$          | 1.88    |

La tasa libre de riesgo es del $2\%$ y el rendimiento esperado del portafolio de mercado es del $8\%$. Con base en el CAPM, ¿cuál es la $\beta$ del portafolio?

A. $\beta_P = 0.95$

B. $\beta_P = 1.19$

**C. $\beta_P = 1.29$**

D. $\beta_P = 1.62$

Considere la siguiente distribución de rendimientos:

| Estado de la economía | Probabilidad       | $R_A$   |
| --------------------- | ------------------ | ------- |
| Depresión             | 0.30               | 0.10    |
| Normal                | 0.50               | 0.15    |
| Expansión             | 0.20               | 0.25    |

Suponiendo que el CAPM es válido, que la volatilidad del portafolio de mercado es $\sigma_M=0.10$ y que la correlación del activo A y el portafolio de mercado es $\rho_{A,M}=0.9$, ¿cuál es el valor de $\beta_A$?

A. $\beta_A=0.25$

B. $\beta_A=0.47$

C. $\beta_A=0.55$

D. $\beta_A=1.15$

Un fondo de inversiones con $\beta=0.8$ tiene un rendimiento esperado del $16\%$. Si la tasa libre de riesgo es $4\%$ y el rendimiento esperado del portafolio de mercado es del $13\%$, ¿debería invertir en este fondo?

**A. Si.**

B. No.

Suponga que encontramos un activo el cual, con base en su precio al día de hoy, tiene un rendimiento esperado más bajo que lo que sugiere la línea de mercado de activos. Si el CAPM es válido, ¿cuál(es) de las siguientes proposiciones es (son) verdaderas?

A. Si el activo tiene un rendimiento esperado más bajo que el sugerido por la línea, significa que su precio es muy bajo.

**B. Si el activo tiene un rendimiento esperado más bajo que el sugerido por la línea, significa que su precio es muy alto.**

C. El activo se grafica por encima de la línea.

**D. El activo se grafica bajo la línea.**

# Anuncios parroquiales

## 1. Último quiz - Lunes 24 de abril
## 2. Revisar archivo Tarea7 - Viernes 21 de abril
## 3. Lunes 24 retornamos a clases presenciales

<script>
  $(document).ready(function(){
    $('div.prompt').hide();
    $('div.back-to-top').hide();
    $('nav#menubar').hide();
    $('.breadcrumb').hide();
    $('.hidden-print').hide();
  });
</script>

<footer id="attribution" style="float:right; color:#808080; background:#fff;">
Created with Jupyter by Esteban Jiménez Rodríguez.
</footer>